component/common.libsonnet

```diff
@@ -19,6 +19,62 @@ local syncOptions = {
   },
 };
 
+local vshnMetaDBaaSExoscale(dbname) = {
+  metadata+: {
+    annotations+: {
+      'metadata.appcat.vshn.io/displayname': 'Exoscale ' + dbname,
+      'metadata.appcat.vshn.io/description': dbname + ' instances hosted by Exoscale',
+      'metadata.appcat.vshn.io/end-user-docs-url': 'https://docs.appuio.cloud/appcat/exoscale-dbaas/' + std.asciiLower(dbname) + '/create.html',
+      'metadata.appcat.vshn.io/zone': 'Exoscale zones: de-fra-1, de-muc-1, at-vie-1, ch-gva-2, ch-dk-2, bg-sof-1',
+      'metadata.appcat.vshn.io/product-description': 'https://products.docs.vshn.ch/products/appcat/exoscale_dbaas.html',
+    },
+    labels+: {
+      'metadata.appcat.vshn.io/offered': 'true',
+      'metadata.appcat.vshn.io/serviceID': 'exoscale-' + std.asciiLower(dbname),
+    },
+  },
+};
+
+local vshnMetaVshn(dbname) = {
+  metadata+: {
+    annotations+: {
+      'metadata.appcat.vshn.io/displayname': 'VSHN Managed ' + dbname,
+      'metadata.appcat.vshn.io/description': dbname + ' instances hosted by VSHN',
+      'metadata.appcat.vshn.io/end-user-docs-url': 'https://docs.appuio.cloud/appcat/vshn-dbaas/' + std.asciiLower(dbname) + '/create.html',
+      'metadata.appcat.vshn.io/zone': 'local cluster',
+      'metadata.appcat.vshn.io/flavor': 'standalone, clustered',
+      'metadata.appcat.vshn.io/product-description': 'https://products.docs.vshn.ch/products/appcat/' + std.asciiLower(dbname) + '.html',
+    },
+    labels+: {
+      'metadata.appcat.vshn.io/offered': 'true',
+      'metadata.appcat.vshn.io/serviceID': 'vshn-' + std.asciiLower(dbname),
+    },
+  },
+};
+
+local vshnMetaSOS(provider) = {
+  metadata+: {
+    annotations+: {
+      'metadata.appcat.vshn.io/displayname': provider + ' ObjectBucket',
+      'metadata.appcat.vshn.io/description': 'S3 compatible object bucket hosted by ' + provider,
+      'metadata.appcat.vshn.io/end-user-docs-url': 'https://docs.appuio.cloud/appcat/object-storage/create.html',
+      'metadata.appcat.vshn.io/zone': provider + ' zones: ' +
+                                      if provider == 'Exoscale' then 'de-fra-1, de-muc-1, at-vie-1, ch-gva-2, ch-dk-2, bg-sof-1' else 'lpg and rma',
```

Comment:
I don't really like this hardcoded list of zones, at least make it a variable to be reused per provider. Soon(ish) we'll provide the service also on other cloud providers, and then we're already better prepared.

Reply:
I defined an array for each one of them.